offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -898,6 +938,192 @@ export class CkbHandler {
+
+    const unlockTxHashes = records
+      .map((unlockRecord) => {
+        return unlockRecord.burnTxHash;
+      })
+      .join(', ');
+
+    logger.info(
+      `CkbHandler doHandleUnlockRecords start process unlock Record, burnTxHashes:${unlockTxHashes} num:${records.length}`,
+    );
+
+    records.map((r) => {
+      r.status = 'pending';
+    });
+    await this.db.saveCollectorCkbUnlock(records);
+    await this.ckbIndexer.waitForSync();
+    const allTxSkeletonWithRecords = await generator.unlock(records);
+    if (!allTxSkeletonWithRecords) {
+      return;
+    }
+    logger.debug(`unlock for records, allTxSkeletonWithRecords`, records, allTxSkeletonWithRecords);
+    logger.debug(
+      `unlock tx txSkeletons: [ ${allTxSkeletonWithRecords
+        .map((txs) => transactionSkeletonToJSON(txs.txSkeleton))
+        .join(', ')} ]`,
+    );
+    for (const txSkeletonWithRecords of allTxSkeletonWithRecords) {
+      for (;;) {
+        try {
+          const txSkeleton = txSkeletonWithRecords.txSkeleton;
+          const records = txSkeletonWithRecords.records;
+          const sigs = await this.collectUnlockSignatures(txSkeleton, records, generator);
+          const multisigs = sigs.join('');
+          logger.info(`multisigs: ${multisigs}`);
+          const smtProof = ForceBridgeCore.getSmtProof();
+          const serializedMultisigScript = serializeMultisigScript(ForceBridgeCore.config.ckb.multisigScript);
+          const signaturePlaceHolder = serializedMultisigScript + multisigs;
+          logger.info(`sigs: ${signaturePlaceHolder}`);
+          const authMultisigBlake160 = new utils.CKBHasher().update(serializedMultisigScript).digestHex().slice(0, 42);
+          const omniLockWitness = {
+            signature: new Reader(signaturePlaceHolder),
+            rc_identity: {
+              identity: new Reader(`0x06${authMultisigBlake160.slice(2)}`),
+              proofs: [{ mask: 3, proof: new Reader(smtProof) }],
+            },
+          };
+          const omniLockWitnessHexString = new Reader(SerializeRcLockWitnessLock(omniLockWitness)).serializeJson();
+          const witness = new Reader(
+            SerializeWitnessArgs(
+              normalizers.NormalizeWitnessArgs({
+                lock: omniLockWitnessHexString,
+              }),
+            ),
+          ).serializeJson();
+          const signedTxSkeleton = txSkeleton.update('witnesses', (witnesses) => {
+            return witnesses.map((value, index) => {
+              if (index === 0) {
+                return witness;
+              }
+              return value;
+            });
+          });
+          const transaction = createTransactionFromSkeleton(signedTxSkeleton);
+          if (sigs.length < ForceBridgeCore.config.ckb.multisigScript.M) {
+            throw new Error(
+              `createBridgeCell collect signatures failed, expected:${ForceBridgeCore.config.ckb.multisigScript.M}, collected:${sigs.length}`,
+            );
+          }
```

Comment:
Move the code up, check sigs.length before construct tx

Reply:
fixed